### PR TITLE
Use several clang-tidy suggestions

### PR DESCRIPTION
**Pull request by FolMing**
So, I was able to run clang-tidy on our codebase.
Some of the changes can be a bit controversial and maybe will require a bit of a talk between us.

### CODE REVIEW

**Review by ArthurBandaryk — APPROVED** [review]

File: eventuals/http.h [hunk 1]
```diff
@@ -958,7 +958,9 @@ struct _HTTP final {
 
               // Transform 'Request' headers to curl's linked list.
               for (const auto& [key, value] : request_.headers()) {
-                std::string header = key + ": " + value;
+                std::string header = key;
```
Comment (CodingCanuck) [inline eventuals/http.h:961]: I'm not in love with this switch to repeated calls to `append`: It seems nicer to instead switch to https://github.com/fmtlib/fmt

**Review by CodingCanuck — APPROVED** [review]
Overall, I like this a lot! I love losing the unneeded `move`s on primitive types.

File: eventuals/http.h (hunk 1 above)
Comment (FolMing) [reply eventuals/http.h:961]: I'm adding a comment for now, but I agree, we should probably use fmt library

**Review by benh — APPROVED** [review]
LGTM!

**Pull request merged**